x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/StringContainsFunctionProcessor.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.expression.function.scalar.string;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.eql.EqlIllegalArgumentException;
+import org.elasticsearch.xpack.ql.expression.gen.processor.Processor;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class StringContainsFunctionProcessor implements Processor {
+
+    public static final String NAME = "sstc";
+
+    private final Processor string, substring;
+
+    public StringContainsFunctionProcessor(Processor string, Processor substring) {
+        this.string = string;
+        this.substring = substring;
+    }
+
+    public StringContainsFunctionProcessor(StreamInput in) throws IOException {
+        string = in.readNamedWriteable(Processor.class);
+        substring = in.readNamedWriteable(Processor.class);
+    }
+
+    @Override
+    public final void writeTo(StreamOutput out) throws IOException {
+        out.writeNamedWriteable(string);
+        out.writeNamedWriteable(substring);
+    }
+
+    @Override
+    public Object process(Object input) {
+        return doProcess(string.process(input), substring.process(input));
+    }
+
+    public static Object doProcess(Object string, Object substring) {
+        if (string == null) {
```

Comment:
```suggestion
        if (string == null || substring == null) {
```

Reply:
throwing exception for now if the second param is null, consistent across all 3 functions that I implemented so far

Reply:
are we learning towards returning nulls more often and not throwing exceptions?